rust-symcrypt/src/ecc/mod.rs

```diff
@@ -0,0 +1,627 @@
+                curve_to_ec_point_format(curve_type),
+                ec_key_usage.to_symcrypt_flag(),
+                key_ptr.0,
+            ) {
+                symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR => {
+                    let key = EcKey {
+                        inner_key: key_ptr,
+                        curve_type: curve_type,
+                        ec_key_usage,
+                        has_private_key: true,
+                    };
+                    Ok(key)
+                }
+                err => Err(err.into()),
+            }
+        }
+    }
+
+    /// `set_public_key()` sets the public key information onto the [`EcKey`].
+    ///
+    /// `curve_type` represents the [`CurveType`] of the key.
+    ///
+    /// `public_key` is a `&[u8]` that represents the public key. This key type must match that of the curve type and will fail with [`SymCryptError::InvalidArgument`] if it does not.
+    ///
+    /// `ec_key_usage` represents the [`EcKeyUsage`] of the key and indicates if the key will be used for [`EcKeyUsage::EcDsa`] or [`EcKeyUsage::EcDh`], or [`EcKeyUsage::EcDhAndEcDsa`].
+    ///
+    ///  This function will return a [`SymCryptError::MemoryAllocationFailure`] if there is not enough memory to allocate the key.
+    pub fn set_public_key(
+        curve_type: CurveType,
+        public_key: &[u8],
+        ec_key_usage: EcKeyUsage,
+    ) -> Result<Self, SymCryptError> {
+        let ec_curve = InnerEcCurve::new(curve_type); // Can fail here due to insufficient memory.
+        unsafe {
+            // SAFETY: FFI calls
+            // Stack allocated since we will do SymCryptEckeyAllocate.
+            let key_ptr = InnerEcKey(symcrypt_sys::SymCryptEckeyAllocate(ec_curve.0));
+            if key_ptr.0.is_null() {
+                return Err(SymCryptError::MemoryAllocationFailure);
+            }
+
+            match symcrypt_sys::SymCryptEckeySetValue(
+                std::ptr::null(), // private key set to null since none is generated
+                0,
+                public_key.as_ptr(), // only a public key is attached
+                public_key.len() as symcrypt_sys::SIZE_T,
+                curve_to_num_format(curve_type),
+                curve_to_ec_point_format(curve_type),
+                ec_key_usage.to_symcrypt_flag(),
+                key_ptr.0,
+            ) {
+                symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR => {
+                    let key = EcKey {
+                        inner_key: key_ptr,
+                        curve_type: curve_type,
+                        ec_key_usage,
+                        has_private_key: false,
+                    };
+                    Ok(key)
+                }
+                err => Err(err.into()),
+            }
+        }
+    }
+
+    /// `export_public_key()` returns the public key associated with the [`EcKey`] as a `Vec<u8>`.
+    pub fn export_public_key(&self) -> Result<Vec<u8>, SymCryptError> {
```

Comment:
>export_public_key [](http://example.com/codeflow?start=11&length=17)

We don't allow export of the private key?

Reply:
Export of the private key was not needed for the verify functions, I'll include it in this PR for completion